cvmfs/swissknife_sync.cc

```diff
@@ -559,6 +560,18 @@ bool swissknife::CommandSync::ReadFileChunkingArgs(
 int swissknife::CommandSync::Main(const swissknife::ArgumentList &args) {
   string start_time = GetGMTimestamp();
 
+  // Spawn monitoring process (watchdog)
+  std::string watchdog_dir = "/tmp";
+  char watchdog_path[PATH_MAX];
+  std::string timestamp = GetGMTimestamp("%Y.%m.%d-%H.%M.%S");
+  int path_size = snprintf(watchdog_path, sizeof(watchdog_path),
+                           "%s/cvmfs-swissknife-sync-stacktrace.%s.%d",
+                           watchdog_dir.c_str(), timestamp.c_str(), getpid());
+  assert(path_size > 0);
+  assert(path_size < PATH_MAX);
+  Watchdog *watchdog = Watchdog::Create(std::string(watchdog_path));
+  watchdog->Spawn();
```

Comment:
I think we need to use a unique pointer here so that the watchdog gets deleted at the end.  Otherwise all terminations, even successful ones, will raise an alarm.